californium-core/src/main/java/org/eclipse/californium/core/coap/Request.java

```diff
@@ -401,11 +404,14 @@ public String getURI() {
 		} else {
 			port = -1;
 		}
-		String path = new StringBuilder("/").append(getOptions().getUriPathString()).toString();
-		String query = getOptions().getUriQueryString();
+		// according RFC7252, section 6.5, item 7, a empty resource name is represented by "/" as path. 
+		// therefore always use the leading "/", even if the uri path is empty. 
+		String path = "/" + getOptions().getUriPathString();
+		String query = getOptions().getURIQueryCount() > 0 ? getOptions().getUriQueryString() : null;
```

Comment:
assuming that this is about preventing to get back just a `?`, wouldn't it make sense to include the check in `Options.getUriQueryString()` itself?